### PR TITLE
First commit - Done with some errors

### PR DESCRIPTION
**Pull request by RVDotsenko**
- [DEMO LINK](https://RVDotsenko.github.io/layout_catalog/)
- [TEST REPORT LINK](https://RVDotsenko.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,8 +5,451 @@
...
+        </a>
+      </div>
+    
+      <nav class="nav">
+        <ul class="nav-list">
+          <li class="nav-value">
+            <a 
+              class="is-active" 
+              href="#Apple"
+            >
```
Comment (polosanya) [inline src/index.html:26]: You can leave it in one line if there are fewer than 3 attributes and they don't take too much width
```suggestion
            <a class="is-active" href="#Apple">
```

File: src/index.html [hunk 2]
```diff
@@ -5,8 +5,451 @@
...
+      </nav>
+    </header>
+    
+    <main class="main">
+      <div class="cards-container">
+        <div class="card-container" data-qa="card">
+          <div class="product-card">
+            <div class="product-image">
+              <img
+                class="product-image__image"
```
Comment (polosanya) [inline src/index.html:97]:
```suggestion
          <div class="card">
            <div class="card__image-container">
              <img
                class="card__image"
```
Please check your BEM structure
It is not correct now
https://mate-academy.github.io/style-guides/htmlcss.html#oocss-and-bem

**Review by polosanya — CHANGES_REQUESTED** [review]
You need to refresh your knowledge of BEM methodology because your BEM structure is incorrect for now. Please check these links:
https://css-tricks.com/bem-101/
https://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/

File: src/index.html [hunk 3]
```diff
@@ -5,8 +5,477 @@
...
     <link rel="stylesheet" href="./styles/main.scss">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
   </head>
   <body>
-    <h1>Catalog</h1>
+    <header class="header">
+      <div class="logo">
+        <a href="#" class="logo__link">
```
Comment (AlisaVlasova) [inline src/index.html:15]:
```suggestion
    <header class="header">
      <div class="header__logo logo">
        <a href="#" class="logo__link">
```

File: src/index.html [hunk 4]
```diff
@@ -5,8 +5,477 @@
...
+          <img src="./images/logo.png" alt="logo-img">
+        </a>
+      </div>
+    
+      <nav class="nav">
+        <ul class="nav-list">
+          <li class="nav-value">
+            <a class="is-active" href="#Apple">
```
Comment (AlisaVlasova) [inline src/index.html:23]:
```suggestion
        <ul class="nav__list">
          <li class="nav__value">
            <a class="nav__link is-active" href="#Apple">
```

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1 @@
-body {
-  margin: 0;
-}
+@import "./blocks/page.scss";
```
[the remark this answers is not in the document]
Comment (AlisaVlasova) [reply src/styles/main.scss:1]: ++

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]

File: src/styles/blocks/header/_logo.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.header__logo {
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/header/_logo.scss:1]: should be in `header.scss`

File: src/styles/blocks/header/_navigation.scss [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  display: flex;
+  list-style: none;
+}
+
+.nav__value:not(:first-child) {
+  margin-left: 20px;
+
+  &:hover {
+    cursor: pointer;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/header/_navigation.scss:13]: link is pointer by default

File: src/styles/blocks/header/_navigation.scss [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+.is-active {
+  @extend %nav-styles;
+  color: $colorForlinksHover;
+  padding-bottom: 0;
+
+  &::after {
+    content: "";
+    position: relative;
+    display: block;
+    top: 19px;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/header/_navigation.scss:34]:
```suggestion
    bottom: 0;
```

File: src/styles/blocks/main/card/_button.scss [hunk 9]
```diff
@@ -0,0 +1,26 @@
+.card {
+  &__container--button {
+    display: flex;
+    text-align: center;
+    cursor: pointer;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_button.scss:5]: link is pointer by default

File: src/styles/blocks/main/card/_button.scss [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  &__button {
+    width: 166px;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_button.scss:9]: don't hardcode width here

File: src/styles/blocks/main/card/_card.scss [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  margin-bottom: 46px;
+
+  padding: 32px $cardLeftPaddings $cardRightPaddings;
+
+  border: $cardBorderWidth solid #f3f3f3;
+  border-radius: 5px;
+
+  background-color: #fff;
+
+  letter-spacing: normal;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_card.scss:18]: do you need this?

File: src/styles/blocks/main/card/_price.scss [hunk 12]
```diff
@@ -0,0 +1,19 @@
+.card {
+  &__container--price {
+    @extend %flex-and-space-between;
+    align-items: center;
+    margin-bottom: 16px;
+  }
+
+  &__price-text {
+    color: $colorSecondary;
+    font-weight: 400;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_price.scss:10]: it's default value

File: src/styles/blocks/main/card/_rating.scss [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+.stars-star {
+  width: 16px;
+  height: 16px;
+  background-repeat: no-repeat;
+  background-position: 50%;
+  margin-right: 4px;
+  background-image: url(../images/star.svg);
+}
+
+.stars--4 > :nth-child(-n + 4) {
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_rating.scss:28]: should work for others numbers of stars

File: src/styles/blocks/main/card/_rating.scss [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+.stars-star {
+  width: 16px;
+  height: 16px;
+  background-repeat: no-repeat;
+  background-position: 50%;
+  margin-right: 4px;
+  background-image: url(../images/star.svg);
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_rating.scss:25]: combine into `background` property

File: src/styles/blocks/main/card/_text.scss [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+  &__text-description {
+    font-size: 12px;
+    line-height: 18px;
+    font-weight: 500;
+  }
+
+  &__text-code {
+    padding-top: 4px;
+    font-size: 10px;
+    font-weight: 400;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/main/card/_text.scss:16]: it's default value

File: src/styles/blocks/page.scss [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+  display: flex;
+  justify-content: center;
+}
+
+* {
+  box-sizing: border-box;
+}
+
+.cards-container {
+  letter-spacing: -0.36em;
```
Comment (ivanchuk-yurii) [inline src/styles/blocks/page.scss:28]: do you need this?

File: src/styles/utils/_variables.scss [hunk 17]
```diff
@@ -0,0 +1,18 @@
+$colorMain: #060b35;
```
Comment (ivanchuk-yurii) [inline src/styles/utils/_variables.scss:1]: use kebab-case for variable naming
```suggestion
$color-main: #060b35;
```

File: src/index.html [hunk 18]
```diff
@@ -5,8 +5,477 @@
...
     <title>Catalog</title>
     <link rel="stylesheet" href="./styles/main.scss">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
   </head>
   <body>
-    <h1>Catalog</h1>
+    <header class="header">
+      <div class="logo">
```
Comment (ivanchuk-yurii) [inline src/index.html:14]: do you need this block?

File: src/index.html (hunk 4 above)
Comment (ivanchuk-yurii) [reply src/index.html:23]: not fixed

File: src/index.html [hunk 19]
```diff
@@ -5,8 +5,477 @@
...
+        <a href="#" class="logo__link">
+          <img src="./images/logo.png" alt="logo-img">
+        </a>
+      </div>
+    
+      <nav class="nav">
+        <ul class="nav-list">
+          <li class="nav-value">
+            <a class="is-active" href="#Apple">
+              apple
```
Comment (ivanchuk-yurii) [inline src/index.html:24]:
```suggestion
              Apple
```
it's better to uppercase first letter

File: src/index.html [hunk 20]
```diff
@@ -5,8 +5,477 @@
...
+            </a>
+          </li>
+        </ul>
+      </nav>
+    </header>
+    
+    <main class="main">
+      <div class="cards-container">
+        <div class="card" data-qa="card">
+          <div class="card__container card__container--image">
```
Comment (ivanchuk-yurii) [inline src/index.html:80]: these `card__container`s are redundant

**Review by ivanchuk-yurii — CHANGES_REQUESTED** [review]
it doesn't display correctly
image

File: src/styles/blocks/main/card/_image.scss [hunk 21]
```diff
@@ -0,0 +1,11 @@
...
+  &__container--image {
+    justify-content: center;
+    margin-bottom: 40px;
+  }
+
+  &__image {
+    width: 160px;
+    height: 134px;
+  }
+}
```
Comment (SanyaBratashchuk) [inline src/styles/blocks/main/card/_image.scss:11]: move all elements of Card BEM blok to the card.scss

File: src/index.html [hunk 22]
```diff
@@ -5,8 +5,477 @@
...
+            </p>
+          </div>
+
+          <div class="card__container card__container--rating">
+            <div class="card__stars stars stars--4">
+              <div class="stars-star"></div>
+              <div class="stars-star"></div>
+              <div class="stars-star"></div>
+              <div class="stars-star"></div>
+              <div class="stars-star"></div>
```
Comment (SanyaBratashchuk) [inline src/index.html:104]:
```suggestion
              <div class="stars__star"></div>
              <div class="stars-star"></div>
              <div class="stars-star"></div>
              <div class="stars-star"></div>
              <div class="stars-star"></div>
```

File: src/styles/blocks/main/card/_rating.scss [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+  height: 16px;
+  margin-right: 4px;
+  background: no-repeat 50% url(../images/star.svg);
+}
+
+@for $index from 1 through $active-stars {
+  .stars--4 > :nth-child(-n + #{$index}) {
+    background-image: url(../images/star-active.svg);
+  }
+}
```
Comment (SanyaBratashchuk) [inline src/styles/blocks/main/card/_rating.scss:30]: Reuse all styles from task with stars and move each BEM block to the separate file.

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/styles/blocks/main/_star.scss [hunk 24]
```diff
@@ -0,0 +1,12 @@
+.stars__star {
```
Comment (DarkMistyRoom) [inline src/styles/blocks/main/_star.scss:1]: Where is stars block? You should have `stars `selector

File: src/styles/blocks/header/_header.scss [hunk 25]
```diff
@@ -0,0 +1,8 @@
+.header {
+  @extend %flex-and-space-between;
+  padding: 0 50px;
+}
+
+.header__logo {
```
Comment (DarkMistyRoom) [inline src/styles/blocks/header/_header.scss:6]: Use nesting

File: src/styles/blocks/header/_navigation.scss [hunk 26]
```diff
@@ -0,0 +1,35 @@
+.nav__list {
```
Comment (DarkMistyRoom) [inline src/styles/blocks/header/_navigation.scss:1]: Where is nav block? I see only elements, but they can't exist without a block, according to BEM

File: src/styles/blocks/main/_star.scss [hunk 27]
```diff
@@ -0,0 +1,12 @@
...
+  width: 16px;
+  height: 16px;
+  margin-right: 4px;
+  background: no-repeat 50% url(../images/star.svg);
+}
+
+@for $index from 1 through $active-stars {
```
Comment (DarkMistyRoom) [inline src/styles/blocks/main/_star.scss:8]: Same comment. You should reuse stars block from the previous task
image

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
The catalog shouldn't go over the left side.
image

**Pull request closed without merging**